hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionFileSystem.java

```diff
@@ -1247,4 +1149,186 @@ private static void sleepBeforeRetry(String msg, int sleepMultiplier, int baseSl
     }
     Thread.sleep((long)baseSleepBeforeRetries * sleepMultiplier);
   }
+
+  private void createWriteStrategy(Configuration conf) {
+    String className = conf.get(REGION_WRITE_STRATEGY, DefaultWriteStrategy.class.getName());
+    try {
+      LOG.info("instantiating write strategy {}", className);
+      writeStrategy = ReflectionUtils.instantiateWithCustomCtor(className,
+        new Class[] { HRegionFileSystem.class }, new Object[] { this });
+    } catch (Exception e) {
+      LOG.error("Unable to create write strategy: {}", className, e);
+    }
+  }
+
+  public static class DefaultWriteStrategy extends HRegionFileSystemWriteStrategy {
+
+    public DefaultWriteStrategy(HRegionFileSystem fileSystem){
+      super(fileSystem);
+    }
+
+    /**
+     * Constructs a Path for the split dir as follows:
+     *  "/hbase/data/NS/TABLE/PARENT_REGION_DIR/.splits/"
+     * @return the temporary parent path for the split dir
+     */
+    @Override
+    public Path getParentSplitsDir() {
+      return new Path(fileSystem.getRegionDir(), REGION_SPLITS_DIR);
+    }
+
+    /**
+     * Constructs a Path for the merged dir as follows:
+     *  "/hbase/data/NS/TABLE/PARENT_REGION_DIR/.merges/"
+     * @return the temporary parent path for the merges dir.
+     */
+    @Override
+    public Path getParentMergesDir() {
+      return new Path(fileSystem.getRegionDir(), REGION_MERGES_DIR);
+    }
+
+    /**
+     * Creates the region splits directory. Assumes getSplitsDir implementation returns a tmp dir,
+     * therefore, it deletes any existing directory returned by getSplitsDir.
+     *
+     * @param daughterA the first half of the split region
+     * @param daughterB the second half of the split region
+     *
+     * @throws IOException if splits dir creation fails.
+     */
+    @Override
+    public void createSplitsDir(RegionInfo daughterA, RegionInfo daughterB) throws IOException {
+      Path splitdir = getParentSplitsDir();
+      if (this.fileSystem.fs.exists(splitdir)) {
+        LOG.info("The " + splitdir + " directory exists.  Hence deleting it to recreate it");
```

Comment:
nit, move over to the slf4j marker message

Reply:
Ack